src/OpenLoco/src/GameCommands/Track/RemoveTrack.cpp

```diff
@@ -0,0 +1,329 @@
+                }
+                if (elSignal->getRight().hasSignal())
+                {
+                    srArgs.flags |= 1U << 14;
+                }
+            }
+
+            if (auto cost = GameCommands::doCommand(srArgs, flags); cost != FAILURE)
+            {
+                totalRemovalCost += cost;
+            }
+            else
+            {
+                return FAILURE;
+            }
+        }
+
+        // 0x0049C967
+        // Fetch track element again; signal removal above might have invalidated the pointer.
+        elTrack = getElTrackAt(args, trackFlags, args.pos, args.index);
+
+        // 0x0049C9F0
+        if (elTrack->hasStationElement())
+        {
+            GameCommands::TrainStationRemovalArgs tsArgs;
+            tsArgs.pos = args.pos;
+            tsArgs.rotation = args.rotation;
+            tsArgs.trackId = args.trackId;
+            tsArgs.index = args.index;
+            tsArgs.type = args.trackObjectId;
+
+            if (auto cost = GameCommands::doCommand(tsArgs, flags); cost != FAILURE)
+            {
+                totalRemovalCost += cost;
+            }
+            else
+            {
+                return FAILURE;
+            }
+        }
+
+        // 0x0049CA3B
+        // Fetch track element again; station removal above might have invalidated the pointer.
+        elTrack = getElTrackAt(args, trackFlags, args.pos, args.index);
+
+        // 0x0049CAC4
+        const auto trackPieces = World::TrackData::getTrackPiece(args.trackId);
+        auto& trackPiece = trackPieces[args.index];
+
+        const auto trackStart = args.pos - World::Pos3{ Math::Vector::rotate(World::Pos2{ trackPiece.x, trackPiece.y }, args.rotation), trackPiece.z };
+
+        // NB: moved out of the loop below (was at 0x0049CC1B)
+        const currency32_t pieceRemovalCost = trackRemoveCost(args, trackPieces[0], trackStart, trackFlags);
+
+        bool trackHadBridge = false; // 0x0113605B
+        int8_t trackBridgeId = -1;   // 0x0113605C
+
+        // 0x0049CB27
+        for (auto& piece : trackPieces)
+        {
+            const auto trackLoc = trackStart + World::Pos3{ Math::Vector::rotate(World::Pos2{ piece.x, piece.y }, args.rotation), piece.z };
+
+            if ((trackFlags & 0x20) == 0) // aiAllocated
+            {
+                World::TileManager::mapInvalidateTileFull(trackLoc); // shift piece.z?
+            }
+
+            // 0x0049CB84
+            auto* pieceElTrack = getElTrackAt(args, trackFlags, trackLoc, piece.index);
+            if (pieceElTrack == nullptr)
+            {
+                return FAILURE;
+            }
+
+            // 0x0049CC02
+            if (pieceElTrack->hasBridge())
+            {
+                trackHadBridge = true;
+                trackBridgeId = pieceElTrack->bridge();
+            }
+
+            if (!(flags & Flags::apply))
+            {
+                continue;
+            }
+
+            auto baseZ = pieceElTrack->baseZ();
```

Comment:
Can we just make it use baseHeight and stick to bigZ in Pos3 coordinates. You would need to modify setLevelCrossingFlags though to use the baseHeight as well